### ISSUE TITLE
Significant differences between output from tests and reference data.

### ISSUE THREAD

**Issue by jjlow** [susilehtola/erkale#9]
I have just built erkale with gcc and gfortran 4.9.3. I used version 2017.0.098 of the mkl math libraries on a sandy bridge processor running linux centos 6 (2.6.32-642.15.1.el6.x86_64).

The results from my build matches most of the references but there are some significant differences. For example, the water dimer tests with wb97x_v and wb97m_v functionals are 11.3 mhartrees lower than the reference.

Could you help me resolve this differences?

**Comment by susilehtola** [comment]
Admittedly I haven't touched the tests in a while, although the defaults of the code have changed so it's not surprising if there are differences. I'll have a look.

**Comment by susilehtola** [comment]
The tests have developed kind of organically over a period of time. I still haven't figured out the best way to redo them.

I have checked the functionals, however, and the results with the current code are correct. I'm not sure what the problem with the tests are.

**Comment by jjlow** [comment]
Susi Lehtola,

I want to use erkale to check the libxc implementation of scan.  I have a set of results for SCAN from Jianwei Sun on his version of Gaussian.  There are significant differences.

John

**Comment by susilehtola** [comment]
SCAN is one of the functionals I've thoroughly tested against Narbe Mardirossian's independent implementation within Q-Chem. Note that the default grid in ERKALE (50,194) is excellent for LDAs and GGAs, but you need to use a larger grid for meta-GGAs such as SCAN.

... however, SCAN is a pathological functional, so you must either look at relative energies, or use ridiculously large grids to make the integration errors small enough. E.g. in my checks with the (50,194) grid absolute energies had discrepancies of 0.1 Eh between ERKALE and Q-Chem, and 1e-4 Eh for relative energies. For (75,302) I got 1e-3 Eh and 1e-4 Eh, for (100,590) 1e-3 Eh and 5e-5 Eh, and for (500,974) 5e-6 Eh and 5e-7 Eh.

tl;dr the functional works, you just need impractically large grids to get a fully converged result.

**Comment by susilehtola** [comment]
Related to #18, I've updated the reference datas for the functionals. The test suite had a really bad grid for meta-GGAs, i.e. just the default one or SG-1 quality.

I checked B97M-V, wB97X-V, and wB97M-V against Q-Chem. The first and last have discrepancies of 10^-7 Eh whereas the middle one for some reason has a discrepancy 10 times larger.

Everything should be okay now.

**Comment by jjlow** [comment]
Susi,

I have been struggling with how to define grids for the SCAN functional.  How do the new grids differ from the old ones?

John J. Low
Principal Computational Science Specialist
Computing, Environment and Life Sciences
Building 240, 2143
9700 South Cass Avenue
Argonne National Laboratory
Argonne, IL 60439.
630-252-0045
www.linkedin.com/pub/john-low/15/8b0/5aa/

**Comment by susilehtola** [comment]
So, before the runfiles didn't specify a DFT grid, which meant that the default (50,194) was used. This is OK for LDAs or GGAs, but for meta-GGAs like B97M-V or wB97M-V you need something bigger, like (75,302) or (99,590). The 

However, SCAN is such a pathological functional that you have to use way bigger grids. I think the main problem in SCAN are the exponential factors in f_x(alpha). If I use a (350,974) grid, I get microhartree agreement with Q-Chem.

So, to run SCAN, in addition to
```
Method mgga_x_scan-mgga_c_scan
```
you should set a large DFT grid such as
```
DFTGrid 350 -974
```
